stackdriver-core/lib/stackdriver/core/configuration.rb

```diff
@@ -22,103 +22,95 @@ module Core
+    #   config = Stackdriver::Core::Configuration.new nested: nested_categories
     #
-    #   config.opt1 = true
-    #   config.category1.opt2 = false
+    #   config.opt1        #=> nil
+    #   config.opt1 = true #=> true
+    #   config.opt1        #=> true
     #
-    #   config.opt1           #=> true
-    #   config.category1.opt2 #=> false
-    #   config.category2.opt3 #=> nil
-    #   config.opt4           #=> RuntimeError: Unrecognized option: opt4
+    #   config.nested1           #=> <Stackdriver::Core::Configuration>
+    #   config.nested2.nested3   #=> <Stackdriver::Core::Configuration>
     #
-    class Configuration
+    class Configuration < Hash
       ##
       # Constructs a new instance of Configuration object.
       #
-      # @param [Array<Symbol, Hash>, Hash<Symbol, Array>] options A nested Array
-      #   or Hash of predefined option keys. Symbols in array will be option
-      #   keys for current level. Nested hash translates to nested Configuration
-      #   objects with nested options.
+      # @param [Symbol, Array<Symbol, Hash>, Hash<Symbol, (Array, Hash)>] nested
+      #   A Symbol, or nested Array and Hash of sub configuration categories.
+      #   A single Symbol and Symbols in array will be keys to next level of
+      #   categories. Nested hash represent sub categories with further nested
+      #   sub categories.
       #
-      def initialize options = []
-        @configs = {}
+      def initialize nested = {}
+        super()
 
-        init_options options
+        add_nested nested
       end
 
       ##
-      # Add more valid options to a Configuration object
+      # Add nested sub configurations to a Configuration object
       #
-      # @param [Array<Symbol, Hash>, Hash<Symbol, Array>] options A nested Array
-      #   or Hash of predefined option keys. Symbols in array will be option
-      #   keys for current level. Nested hash translates to nested Configuration
-      #   objects with nested options.
+      # @param [Symbol, Array<Symbol, Hash>, Hash<Symbol, (Array, Hash)>] nested
+      #   A Symbol, or nested Array and Hash of sub configuration categories.
+      #   A single Symbol and Symbols in array will be keys to next level of
+      #   categories. Nested hash represent sub categories with further nested
+      #   sub categories.
       #
       # @example
-      #   config = Stackdriver::Core::Configuration.new options: [:opt1]
-      #   config.opt2 #=> RuntimeError: Unrecognized option: opt2
+      #   config = Stackdriver::Core::Configuration.new
+      #   config.nest1 #=> nil
+      #   config.add_nested {nest1: [nest2]}
+      #   config.nest1 #=> <Stackdriver::Core::Configuration>
+      #   config.nest1.nest2 #=> <Stackdriver::Core::Configuration>
       #
-      #   config.add_otpions options: [:opt2]
-      #   config.opt2 #=> nil
-      #
-      def add_options options
-        init_options options
+      def add_nested nested
```

Comment:
I think `add_option` or `add_options` is fine. Perhaps rename the argument to something `definition`?

```ruby
def add_options definition
```